### DISCUSSION TITLE
Não funciona em aplicações CONSOLE

### DISCUSSION THREAD

**Discussion by oilujrasec** [Daniel09Fernandes/Component_OpenOffice#63] [General]
Tenho um webservice, e tive a necessidade de gerar uma planilha no servidor e repassar pela API, porém o componente simplismente não funcionar no datamodule, retorna um erro no seu create. Reparei que nos componente para Office que testei tive a mesma situação.

Esse erro:
"Cannot create instance of class CLSID_WICImagingFactory (800401F0)"

**Comment by Daniel09Fernandes** [comment]
Boa tarde @oilujrasec, este erro é relacionado ao WICImagingFactory e não ao componente DinosOffice, experimente colocar no seu Datamodule o codigo: CoInitialize(nil); da uses Winapi.ActiveX, possivelmente no createDataModule deve resolver

**Reply by oilujrasec** [reply]
Já chegou a testar o componente em app console, funcionou?

Eu ja tinha feito isso tb

**Reply by Daniel09Fernandes** [reply]
Bom dia @oilujrasec 
No App console, funciona certinho, mas testei na com uma API Horse e vi que há um problema com o mult Thread, realizei a correção no create do componente, para trabalhar com múltiplas threads e subi um exemplo do uso com o Horse, mas funciona para qualquer ambiente mult threads. 

Segue o exemplo: 
https://github.com/Daniel09Fernandes/DinosOffice/tree/versionOficial/samples/APIHorse

De brinde, subi um método, para transformar a planilha em base64 para devolver da API para o cliente.

**Reply by oilujrasec** [reply]
Ficou lindo, vlw

**Reply by Daniel09Fernandes** [reply]
Aeee Valeu @oilujrasec
